core/src/main/java/org/lionweb/lioncore/java/serialization/JsonSerialization.java

```diff
@@ -80,373 +81,158 @@ public PrimitiveValuesSerialization getPrimitiveValuesSerialization() {
-            return nodeData;
-        } else {
+    private String serializePropertyValue(DataType dataType, Object value) {
+        if (value == null) {
             return null;
         }
+        return primitiveValuesSerialization.serialize(dataType.getID(), value);
     }
 
-    private String getStringProperty(JsonObject jsonObject, String propertyName) {
-        if (!jsonObject.has(propertyName)) {
-            throw new IllegalArgumentException(propertyName + " property not found in " + jsonObject);
-        }
-        JsonElement value = jsonObject.get(propertyName);
-        if (value.isJsonPrimitive() && value.getAsJsonPrimitive().isString()) {
-            return value.getAsJsonPrimitive().getAsString();
-        } else {
-            throw new IllegalArgumentException(propertyName + " property expected to be a string while it is " + value);
+    private void validateSerializationBlock(SerializedChunk serializationBlock) {
+        if (!serializationBlock.getSerializationFormatVersion().equals("1")) {
+            throw new IllegalArgumentException("Only serializationFormatVersion = '1' is supported");
         }
     }
 
-    @Nullable
-    private String tryToGetStringProperty(JsonObject jsonObject, String propertyName) {
-        if (!jsonObject.has(propertyName)) {
-            return null;
-        }
-        JsonElement value = jsonObject.get(propertyName);
-        if (value.isJsonPrimitive() && value.getAsJsonPrimitive().isString()) {
-            return value.getAsJsonPrimitive().getAsString();
-        } else {
-            return null;
+    private List<Node> unserializeSerializationBlock(SerializedChunk serializationBlock) {
+        List<Node> nodes = serializationBlock.getNodes().stream().map(n -> instantiateNodeFromSerialized(n)).collect(Collectors.toList());
+        NodeResolver nodeResolver = new CombinedNodeResolver(new LocalNodeResolver(nodes), this.nodeResolver);
+        serializationBlock.getNodes().stream().forEach(n -> populateNode(n, nodeResolver));
+        return nodes;
+    }
+
+    private Node instantiateNodeFromSerialized(SerializedNode serializedNode) {
+        Concept concept = getConceptResolver().resolveConcept(serializedNode.getConcept());
+        Node node = getNodeInstantiator().instantiate(concept, serializedNode);
+        serializedNode.getProperties().forEach(serializedPropertyValue ->{
+            Property property = concept.getPropertyByMetaPointer(serializedPropertyValue.getMetaPointer());
+            Objects.requireNonNull(property, "Property with metaPointer " + serializedPropertyValue.getMetaPointer() + " not found");
+            Object unserializedValue = primitiveValuesSerialization.unserialize(property.getType().getID(), serializedPropertyValue.getValue());
+            node.setPropertyValue(property, unserializedValue);
+        });
+        return node;
+    }
+
+    private void populateNode(SerializedNode serializedNode, NodeResolver nodeResolver) {
```

Comment:
to my taste this method is a bit long, we could split it up

Reply:
I have now extracted two methods out of this one